backend/routes/api/users.js

```diff
@@ -0,0 +1,166 @@
+const express = require("express");
+const router = express.Router();
+const User = require("../../models/User");
+const { celebrate, Joi } = require("celebrate");
+// get all users
+router.get(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      oauthId: Joi.string().required(),
+      propicUrl: Joi.string().required(),
+      isApproved: Joi.boolean().default(false),
+      role: Joi.string().required(),
+      location: Joi.string().required()
+    })
+  }),
+  async (req, res) => {
+    const users = await User.find({});
+    res.json({
+      code: 200,
+      result: users,
+      success: true
+    });
+  }
+);
+
+// get one user
+router.get("/:user_id", async (req, res) => {
+  const userId = req.params.user_id;
+  const user = await User.findById(userId);
+  res.json({
+    code: 200,
+    result: user,
+    success: true
+  });
+});
+
+// create new user
+router.post(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      oauthId: Joi.string().required(),
+      propicUrl: Joi.string().required(),
+      isApproved: Joi.boolean().default(false),
+      role: Joi.string().required(),
+      location: Joi.string().required()
+    })
+  }),
+  async (req, res) => {
+    const data = req.body;
+    const newUser = new User({
+      firstName: data.firstName,
+      lastName: data.lastName,
+      oauthId: data.oauthId,
+      propicUrl: data.propicUrl,
+      role: data.role,
+      location: data.location
+    });
+    await newUser.save();
+    res.json({
+      code: 200,
+      message: "User Successfully Created",
+      success: true
+    });
+  }
+);
+
+// set role
+router.put(
+  "/setRole/:user_id",
```

Comment:
Super close! This is a nit, but let's rename this to `/users/:user_id/role` to keep consistent with REST standards.